File: src/app/features/vendor-document/vendor-document.ts
import { Component } from '@angular/core';
import { CommonModule } from '@angular/common';
import { Router, RouterLink } from '@angular/router';

interface VendorDocument {
  name: string;
  file: File | null;
  fileName: string;
  size: string;
}

@Component({
  selector: 'app-vendor-document',
  standalone: true,
  imports: [CommonModule, RouterLink],
  templateUrl: './vendor-document.html',
  styleUrls: ['./vendor-document.css']
})
export class VendorDocumentComponent {

  constructor(private router: Router) {}

  documents: VendorDocument[] = [
    {
      name: 'GST Certificate',
      file: null,
      fileName: '',
      size: ''
    },
    {
      name: 'PAN Card',
      file: null,
      fileName: '',
      size: ''
    },
    {
      name: 'Company Registration Certificate',
      file: null,
      fileName: '',
      size: ''
    },
    {
      name: 'ISO Certificate',
      file: null,
      fileName: '',
      size: ''
    },
    {
      name: 'Other Supporting Documents',
      file: null,
      fileName: '',
      size: ''
    }
  ];

  uploadFile(event: Event, index: number): void {

    const input = event.target as HTMLInputElement;

    if (input.files && input.files.length > 0) {

      const file = input.files[0];

      const allowedTypes = [
        'application/pdf',
        'image/jpeg',
        'image/jpg',
        'image/png'
      ];

      if (!allowedTypes.includes(file.type)) {
        alert('Only PDF, JPG, JPEG and PNG files are allowed.');
        input.value = '';
        return;
      }

      const maxSize = 5 * 1024 * 1024;

      if (file.size > maxSize) {
        alert('File size should not exceed 5 MB.');
        input.value = '';
        return;
      }

      this.documents[index].file = file;
      this.documents[index].fileName = file.name;
      this.documents[index].size =
        (file.size / 1024 / 1024).toFixed(2) + ' MB';
    }
  }

  viewFile(index: number): void {

    if (this.documents[index].file) {

      const url = URL.createObjectURL(this.documents[index].file!);
      window.open(url, '_blank');

    } else {

      alert('No file uploaded.');

    }

  }

  deleteFile(index: number): void {

    this.documents[index].file = null;
    this.documents[index].fileName = '';
    this.documents[index].size = '';

    alert('File deleted successfully.');

  }

  save(): void {

    alert('Documents saved successfully.');

  }

  finish(): void {

    alert('Vendor registration completed.');
    this.router.navigate(['/vendor-list']);

  }

}

File: backend/app/models/vendor_document.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey, Boolean
from datetime import datetime
from app.core.database import Base


class VendorDocument(Base):
    __tablename__ = "vendor_documents"

    id = Column(Integer, primary_key=True, index=True)
    vendor_id = Column(Integer, ForeignKey("vendors.id"), nullable=False)

    document_type = Column(String(100), nullable=False)
    # Allowed: GST Certificate, PAN Card, Company Registration Certificate,
    # ISO Certificate, Other Supporting Document

    file_name = Column(String(255), nullable=False)
    file_path = Column(String(500), nullable=False)  # storage location/URL
    file_size = Column(Integer, nullable=True)
    content_type = Column(String(100), nullable=True)  # e.g. application/pdf, image/png

    uploaded_by = Column(Integer, ForeignKey("users.id"), nullable=True)
    uploaded_at = Column(DateTime, default=datetime.utcnow)

    version = Column(Integer, default=1, nullable=False)
    is_current = Column(Boolean, default=True, nullable=False)
    replaced_document_id = Column(Integer, ForeignKey("vendor_documents.id"), nullable=True)
    replaced_at = Column(DateTime, nullable=True)
    replaced_by = Column(Integer, ForeignKey("users.id"), nullable=True)